tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,266 @@
+  std::array<DataVector, 4> char_speeds_five{
+      {get(five), get(five), get(five), get(five)}};
+  std::array<DataVector, 4> char_speeds_minus_five{
+      {get(minus_five), get(minus_five), get(minus_five), get(minus_five)}};
+
+  GeneralizedHarmonic::UpwindFlux<spatial_dim> flux_computer{};
+
+  INFO("test generalized-harmonic upwind weighting function")
+  // If all the char speeds are +1, the weighted fields should just
+  // be the interior fields
+  const auto weighted_char_fields_one =
+      GeneralizedHarmonic::GeneralizedHarmonic_detail::weight_char_fields<3>(
+          char_fields_int, char_speeds_one, char_fields_ext, char_speeds_one);
+  CHECK(weighted_char_fields_one == char_fields_int);
+
+  // If all the char speeds are -1, the weighted fields should just be
+  // the exterior fields up to a sign
+  const auto weighted_char_fields_minus_one =
+      GeneralizedHarmonic::GeneralizedHarmonic_detail::weight_char_fields<3>(
+          char_fields_int, char_speeds_minus_one, char_fields_ext,
+          char_speeds_minus_one);
+  CHECK(weighted_char_fields_minus_one == -1.0 * char_fields_ext);
+
+  // Check scaling by 5 instead of 1
+  const auto weighted_char_fields_minus_five =
+      GeneralizedHarmonic::GeneralizedHarmonic_detail::weight_char_fields<3>(
+          char_fields_int, char_speeds_minus_five, char_fields_ext,
+          char_speeds_minus_five);
+  const auto weighted_char_fields_five =
+      GeneralizedHarmonic::GeneralizedHarmonic_detail::weight_char_fields<3>(
+          char_fields_int, char_speeds_five, char_fields_ext, char_speeds_five);
+  CHECK(weighted_char_fields_minus_five == -5.0 * char_fields_ext);
+  CHECK(weighted_char_fields_five == 5.0 * char_fields_int);
+
+  INFO("test consistency of the generalized-harmonic upwind flux")
+  auto packaged_data_int = make_with_value<Variables<
+      typename GeneralizedHarmonic::UpwindFlux<spatial_dim>::package_tags>>(
+      used_for_size, std::numeric_limits<double>::signaling_NaN());
+  flux_computer.package_data(make_not_null(&packaged_data_int),
+                             spacetime_metric_int, pi_int, phi_int, lapse_int,
+                             shift_int, inverse_spatial_metric_int, gamma_1,
+                             gamma_2, unit_normal_one_form_int);
+
+  auto packaged_data_int_opposite_normal = make_with_value<Variables<
+      typename GeneralizedHarmonic::UpwindFlux<spatial_dim>::package_tags>>(
+      used_for_size, std::numeric_limits<double>::signaling_NaN());
+  flux_computer.package_data(make_not_null(&packaged_data_int_opposite_normal),
+                             spacetime_metric_int, pi_int, phi_int, lapse_int,
+                             shift_int, inverse_spatial_metric_int, gamma_1,
+                             gamma_2, minus_unit_normal_one_form_int);
+  tnsr::i<DataVector, spatial_dim, Frame::Inertial>
+      minus_unit_normal_one_form_ext = unit_normal_one_form_ext;
```

Comment:
this is unused.
After removal `unit_normal_one_form_ext` should also become unused.